utilities/plotting/forcing_animation.py

```diff
@@ -0,0 +1,221 @@
+
+g_commands = commands.add_parser("grid")
+g_commands.add_argument('data', metavar="gridded netcdf precip", type=argparse.FileType('r'))
+
+args = parser.parse_args()
+
+forecast = args.cycle+" Range "+args.source
+crs = ''
+epsg = ''
+if(args.command == 'hydrofabric'):
+    epsg = '5070'
+    crs = ccrs.epsg(epsg)
+    crs = ccrs.PlateCarree()
+elif(args.command == 'grid'):
+    epsg = '4326'
+    # doesn't work: crs= ccrs.epsg('4326')
+    crs = ccrs.PlateCarree() #NJF why this one???
+
+forcing = xr.open_dataset(args.data.name, decode_cf=False)
+forcing.time.attrs['units'] = forcing.time.attrs['reference_time']
+forcing = xr.decode_cf(forcing)
+
+###Setup general figure
+#cmap = sns.color_palette('flare', as_cmap=True)
+#Single hughe blue, really light on the 0 side...
+#cmap = sns.color_palette('Blues', as_cmap=True)
+#Little better constrast, still high desat on the low end
+cmap = sns.color_palette("light:b", as_cmap=True)
+#Reasonably good "blue ish" cube helix
+#cmap = sns.color_palette("ch:start=.2,rot=-.2", as_cmap=True)
+
+bkgrnd = sns.color_palette('flare', as_cmap=True)
+#cmap.set_under(bkgrnd(50), alpha=0.05)
+cmap.set_under("bisque", alpha=0.1)
+e_color = 'none'
+vmin = forcing['RAINRATE'].min().item()
+vmin = 0.0009
+vmax = forcing['RAINRATE'].max().item()
+
+fig = plt.figure(figsize=(19.20, 10.80), dpi=100)
+ax = plt.axes(projection=crs)
+#plt.suptitle("Short Range")
+if(args.switch):
+    cax = fig.add_axes([0.9, 0.1, 0.03, 0.8]) #vertical
+    orient = "vertical"
+else:
+    cax = fig.add_axes([0.05, 0.05, 0.9, 0.03]) #horizontal
+    orient = "horizontal"
+    norm = plt.Normalize(vmin=vmin, vmax=vmax)
+sm = plt.cm.ScalarMappable(cmap=cmap, norm=norm)
+# fake up the array of the scalar mappable. Urgh...
+sm._A = []
+
+cbar = fig.colorbar(sm, cax=cax, orientation=orient)
+cbar.set_label(forcing['RAINRATE'].attrs['units'])
+#ax = fig.gca()
+ax.set_axis_off()
+ax.title.set_size(20)
+
+interval = 100
+fps = 20
+if(args.cycle == "Short"):
+    interval = 5
+    fps = 10
+
+if(args.command == "hydrofabric"):
+    fs = forcing["catchment_ids"].to_series()
+    print( fs[ fs.duplicated() ].values)
+    # WAY faster reading with pyogrio and use_arrow (though just pyogrio is a significant imporvement!)
+    catchments = gpd.read_file(args.hydrofabric.name, layer="divides", include_fields=['divide_id'], engine='pyogrio', use_arrow=True)
+    # union = gpd.GeoSeries( catchments.unary_union, crs=catchments.crs)
+    union = gpd.read_file("domain.gpkg", engine='pyogrio')
+    
+    print("Projecting domain to epsg:4326")
+    union = union.to_crs(epsg='4326')
+    catchments = catchments.to_crs(epsg='4326')
+
+    forcing = forcing.set_index({"catchment ids":"catchment_ids"}).to_array().squeeze().drop_vars('variable')
+    #Get the catchment geopandas index, strip to int for xarray index
+    cat_ids = catchments['divide_id'].apply(lambda x: int(x.split("-")[1]))
```

Comment:
Presumably true at the time of writing, but does the hydrofabric work guarantee unique numeric IDs of catchments across the various kinds that potentially prefix them?

Actually, are there currently or plans for catchment IDs prefixed other than `cat-NNNN`, like we have for nexuses?

@mikejohnson51